src/EC2Store.js

```diff
@@ -97,7 +97,109 @@ class EC2Store {
 	// Should be mapped to a format we expect. i.e.:
 	// { VolumeId, Name, BackupConfig }
 	listEBS () {
-		return new Promise((res) => {res([]);});
+		return new Promise(function(resolve, reject){
+
+			let ec2 = new AWS.EC2();
+
+			ec2.describeVolumes({}, function(error, response){
+				if (error) {
+					reject(error);
+				} else {
+
+					var volumesForBackup = response.Volumes.filter(function(volumes){
+						for (var i=0; i<volumes.Tags.length; i++) {
+							if (volumes.Tags[i].Key === 'backups:config-v0') {
+								return true;
+							} else {
+								return false;
+							}
+						}
+					});
+
+					resolve(volumesForBackup.map(function(volumes){
+
+						var filteredForName = volumes.Tags.filter(function(tag){
+							if (tag.Key === 'Name') {
+								return true;
+							} else {
+								return false;
+							}
+						});
+
+						if (filteredForName.length === 1) {
+							filteredForName = filteredForName[0].Value;
+						} else {
+							throw new Error ('expected to receive volume with single value for name but length > 1');
+						}
+
+						var backupTag = volumes.Tags.filter(function(tag){
+							if ((/[[\d\|\d\]]/g).test(tag.Value) === true){
+								return true;
+							} else {
+								return false;
+							}
+						});
+
+						if (backupTag.length === 1) {
+							backupTag = backupTag[0];
+						} else {
+							throw new Error ('expected to receive volume with single tag for backup instructions, however length > 1');
+						}
+
+						var tagOnly = backupTag.Value.replace(/[\s]/g,'');
+						var tagSplit = tagOnly.split(/,/);
+
+						var aliasArray = {
+							Hourly: [1, 24],
+							Daily: [24, 168],
+							Weekly: [168, 672],
+							Monthly: [672, 8760],
+							Yearly: [8064, 61320]
+						};
+
+						var finalBackupTag = tagSplit.map(function(tag){
+							if ((/[\d]/).test(tag) === true){
+
+								var tupleObj = {
+									Expiry: parseInt(tag.split('|')[1].replace(/[[\]]/g,'')),
+									Frequency: parseInt(tag.split('|')[0].replace(/[[\]]/g,''))
+								};
+
+								return tupleObj; //I've assumed that the tuple will always have two values and be in the correct order, I could write it so that it tests which value is larger to guarantee the larger one becomes Expiry
```

Comment:
You're assumption is correct! 

As stated in the Tag API doc: the first number will always be Frequency and the second number will always be Expiry. This does not necessarily mean that the larger number is the Expiry though. A person using this project might want to make a backup every 12 hours but have it deleted after 2 hours (I have no idea why they would want to do that but it's not up to us to restrict them :stuck_out_tongue:)